applications/sintering/include/pf-applications/sintering/projection.h

```diff
@@ -0,0 +1,210 @@
+      normal[direction] = 1;
+
+      std::vector<unsigned int> projector;
+      for (unsigned int d = 0; d < dim; ++d)
+        if (d != direction)
+          projector.push_back(d);
+
+      auto projection =
+        std::make_unique<StateData<dim - 1, typename VectorType::value_type>>(
+          vector_to_be_used->n_blocks());
+
+      for (const auto &cell :
+           background_dof_handler_to_be_used->active_cell_iterators())
+        if (cell->is_locally_owned())
+          {
+            auto ref_point       = cell->center();
+            ref_point[direction] = location;
+
+            if (cell->bounding_box().point_inside(ref_point))
+              {
+                CellData<dim - 1> cell_data;
+                unsigned int      vertex_counter = 0;
+                unsigned int vertex_numerator = projection->solution[0].size();
+
+                for (unsigned int b = 0; b < vector_to_be_used->n_blocks(); ++b)
+                  {
+                    projection->solution[b].grow_or_shrink(
+                      projection->solution[b].size() +
+                      cell_data.vertices.size());
+                  }
+
+                // Iterate over each line of the cell
+                for (unsigned int il = 0; il < cell->n_lines(); il++)
+                  {
+                    // Check if there are intersections with box planes
+                    const auto [has_itersection, fac, p] =
+                      intersect_line_plane(cell->line(il)->vertex(0),
+                                           cell->line(il)->vertex(1),
+                                           origin,
+                                           normal);
+
+                    if (has_itersection && std::abs(fac) < 1. + tolerance)
+                      {
+                        cell_data.vertices[vertex_counter] = vertex_numerator;
+
+                        Point<dim - 1> p_proj;
+                        for (unsigned int j = 0; j < projector.size(); ++j)
+                          {
+                            p_proj[j] = p[projector[j]];
+                          }
+                        vertices.push_back(p_proj);
+
+                        // Now interpolate values
+                        // DOFs correspnding to the vertices
+                        const auto index0 =
+                          cell->line(il)->vertex_dof_index(0, 0);
+                        const auto index1 =
+                          cell->line(il)->vertex_dof_index(1, 0);
+
+                        for (unsigned int b = 0;
+                             b < vector_to_be_used->n_blocks();
+                             ++b)
+                          {
+                            // The field values associated with those DOFs
+                            const auto val0 = vector.block(b)[index0];
+                            const auto val1 = vector.block(b)[index1];
+
+                            const auto val_proj = val0 + fac * (val0 - val1);
```

Comment:
We need to remember that this only works for linear elements. Maybe you could add an assert?

Reply:
Sure, that makes sense. I'll add one.